src/Core/AdminConsole/OrganizationFeatures/OrganizationUsers/RemoveOrganizationUserCommand.cs

```diff
@@ -8,38 +10,171 @@ namespace Bit.Core.AdminConsole.OrganizationFeatures.OrganizationUsers;
+        {
+            try
+            {
+                if (deletingUserId.HasValue && orgUser.UserId == deletingUserId)
+                {
+                    throw new BadRequestException("You cannot remove yourself.");
+                }
+
+                if (orgUser.Type == OrganizationUserType.Owner && deletingUserId.HasValue && !deletingUserIsOwner)
+                {
+                    throw new BadRequestException("Only owners can delete other owners.");
+                }
 
-        await _organizationService.RemoveUserAsync(organizationId, organizationUserId, eventSystemUser);
+                await _eventService.LogOrganizationUserEventAsync(orgUser, EventType.OrganizationUser_Removed);
+
+                if (orgUser.UserId.HasValue)
+                {
+                    await DeleteAndPushUserRegistrationAsync(organizationId, orgUser.UserId.Value);
+                }
+                result.Add(Tuple.Create(orgUser, ""));
+                deletedUserIds.Add(orgUser.Id);
+            }
+            catch (BadRequestException e)
+            {
+                result.Add(Tuple.Create(orgUser, e.Message));
+            }
+
+            await _organizationUserRepository.DeleteManyAsync(deletedUserIds);
+        }
+
+        return result;
     }
 
-    private async Task ValidateDeleteUserAsync(Guid organizationId, Guid organizationUserId)
+    private void ValidateDeleteUser(Guid organizationId, OrganizationUser orgUser)
     {
-        var orgUser = await _organizationUserRepository.GetByIdAsync(organizationUserId);
         if (orgUser == null || orgUser.OrganizationId != organizationId)
         {
             throw new NotFoundException("User not found.");
         }
     }
+
+    private async Task RepositoryDeleteUserAsync(OrganizationUser orgUser, Guid? deletingUserId)
+    {
+        if (deletingUserId.HasValue && orgUser.UserId == deletingUserId.Value)
+        {
+            throw new BadRequestException("You cannot remove yourself.");
+        }
+
+        if (orgUser.Type == OrganizationUserType.Owner)
+        {
+            if (deletingUserId.HasValue && !await _currentContext.OrganizationOwner(orgUser.OrganizationId))
+            {
+                throw new BadRequestException("Only owners can delete other owners.");
+            }
+
+            if (!await _hasConfirmedOwnersExceptQuery.HasConfirmedOwnersExceptAsync(orgUser.OrganizationId, new[] { orgUser.Id }, includeProvider: true))
+            {
+                throw new BadRequestException("Organization must have at least one confirmed owner.");
+            }
+        }
+
+        await _organizationUserRepository.DeleteAsync(orgUser);
+
+        if (orgUser.UserId.HasValue)
+        {
+            await DeleteAndPushUserRegistrationAsync(orgUser.OrganizationId, orgUser.UserId.Value);
+        }
+    }
+
+    private async Task<IEnumerable<KeyValuePair<string, DeviceType>>> GetUserDeviceIdsAsync(Guid userId)
```

Comment:
I have copied these methods from `OrganizationService`. Should I create a new command for this? Or is too much scope creep at this point?

Reply:
Probably too much at this point.